### DISCUSSION TITLE
no org id

### DISCUSSION THREAD

**Discussion by yangzw1024** [grafana/mimir#2322] [Help and support] [answered]
I'm used grafana mimir without an authenticating, and it's auto set the tenant ID to the string anonymous, when I want to query some metrics from the query-frontend useing url `http://<query-frontend>:8080/api/v1/query`, it's returned `no org id`, I'm changed the url to `http://<query-frontend>:8080/anonymous/api/v1/query`, then it's returnd 404,  which is the right url to query metrics?

**Comment by rlex** [comment, marked as answer]
AFAIK you still need X-Scope-OrgID header set even with auth disabled and URL should be same for every tenant (without /anonymous/ part)
From mimir-nginx gateway:

```
  # Ensure that X-Scope-OrgID is always present, default to "anonymous" for backwards compatibility when multi-tenancy was turned off.
  map $http_x_scope_orgid $ensured_x_scope_orgid {
    default $http_x_scope_orgid;
    "" "anonymous";
  }

proxy_set_header X-Scope-OrgID $ensured_x_scope_orgid;
```

**Reply by pracucci** [reply]
That's correct.

Multi-tenancy is enabled by default and can be disabled via `-auth.multitenancy-enabled=false`. If you keep multi-tenancy enabled, then every API request is required to have the `X-Scope-OrgID` header with the value set to the tenant ID you're authenticating for (`anonymous` is the default tenant ID used when multi tenancy is disabled).

You can learn more in [Grafana Mimir authentication and authorization](https://grafana.com/docs/mimir/next/operators-guide/securing/authentication-and-authorization/).

**Comment by farisamour** [comment]
So basically you either disable multi tenancy or send the X-scope-orgid in the header, in my case i wanted to finish a poc, so i wanted to disable the multi tenancy, disabling it on global level yaml file didnt work, so i disabled it component by component in the extraArgs like below:

```
ruler:
  extraArgs:
    auth.multitenancy-enabled: false
```

ref: https://grafana.com/docs/mimir/v2.1.x/operators-guide/securing/authentication-and-authorization/
